pandas/io/formats/style_render.py

```diff
@@ -1203,6 +1207,17 @@ def format_index(
             Convert string patterns containing https://, http://, ftp:// or www. to
             HTML <a> tags as clickable URL hyperlinks if "html", or LaTeX \href
             commands if "latex".
+        rename : list of str, list of list of str
```

Comment:
umm the user can simply `.rename_axis` on the frame no? why adding api here that is duplicative

Reply:
my design objective with styler has been to customise the display of tables without altering the underlying dataframe object (which is of course possible to get around many of the styler functions).

you can lose the power of indexing if you relabel for the purpose of printout.

additionally, methods like `DataFrame.to_html` and `DataFrame.to_latex` use a `headers` argument which does this relabelling. So in order to replicate that behaviour with the styler code need this pr.

Reply:
here's a good example:
![Screenshot 2022-03-07 at 18 14 58](https://user-images.githubusercontent.com/24256554/157084175-a783dfdb-ae05-4591-b62f-bf719b4606b9.png)